pkg/checktypes/code.go

```diff
@@ -0,0 +1,226 @@
+	if err != nil {
+		return "", err
+	}
+	if !modified {
+		logger.Infof("No changes in checktype in dir %s, reusing image %s", string(c), c.imageName())
+		return c.imageName(), nil
+	}
+	logger.Infof("Compiling checktype in dir %s", c)
+	dir := string(c)
+	// Run go build in the checktype dir.
+	if err := goBuildDir(dir); err != nil {
+		return "", err
+	}
+	// Build a Tar file with the docker image contents.
+	logger.Infof("Building image for checktype in dir %s", dir)
+	contents, err := buildTarFromDir(dir)
+	if err != nil {
+		return "", err
+	}
+	logger.Debugf("Tar file for checktype in dir %s built", dir)
+	modif, err := c.lastModified(logger)
+	if err != nil {
+		return "", err
+	}
+	t := modif.Format(time.RFC822)
+	logger.Debugf("Last modified time for checktype in dir %s is %s", dir, t)
+	labels := map[string]string{modifTimeLabel: t}
+	image := c.imageName()
+	r, err := buildDockerdImage(contents, []string{image}, labels)
+	if err != nil {
+		return "", err
+	}
+	logger.Infof("Docker image built: %s", image)
+	logger.Debugf("Docker image build log:\n%s", r)
+	return image, nil
+}
+
+func (c Code) isModified(logger log.Logger) (bool, error) {
+	labels, err := imageInfo(c.imageName())
+	if err != nil {
+		return false, err
+	}
+	imageTimeS, ok := labels[modifTimeLabel]
+	if !ok {
+		logger.Infof("Image %s is does not contain the label %s", c.imageName(), modifTimeLabel)
+		return true, nil
+	}
+	_, err = time.Parse(time.RFC822, imageTimeS)
+	if err != nil {
+		logger.Infof("invalid time, %+w defined in the label %s of the image %s", err, modifTimeLabel, c.imageName())
+		return true, nil
+	}
+	dirTime, err := c.lastModified(logger)
+	if err != nil {
+		err := fmt.Errorf("error: %+w, getting the last modification time for the checktype in %s", err, string(c))
+		return false, err
+	}
+	dirTimeS := dirTime.Format(time.RFC822)
+	logger.Debugf("Last modified time in dir %s: %s, image time: %s ", c, imageTimeS, dirTimeS)
+	modified := dirTimeS != imageTimeS
+	return modified, nil
+}
+
+func (c Code) lastModified(logger log.Logger) (time.Time, error) {
+	dir := string(c)
+	var latest *time.Time
+	logger.Debugf("Getting the last modified file in dir %s", c)
+	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		logger.Debugf("Visiting path %s", path)
+		if d.IsDir() {
+			if d.Name() == ".git" {
+				return filepath.SkipDir
+			}
+			return nil
+		}
+		info, err := d.Info()
+		if err != nil {
+			return err
+		}
+		modtime := info.ModTime()
+		if latest == nil || modtime.After(*latest) {
+			latest = &modtime
+		}
+		return nil
+	})
+	if err != nil {
+		return time.Time{}, fmt.Errorf("error walking through the dir %s", dir)
+	}
+	if latest == nil {
+		return time.Time{}, fmt.Errorf("the dir %s is empty", dir)
+	}
+	return *latest, nil
+}
+
+func (c Code) imageName() string {
+	dir := string(c)
+	image := path.Base(dir)
+	return fmt.Sprintf("%s-%s", image, "local")
```

Comment:
In case of multiple checktypes it would lead in name collision if we use the same checktype-name 

```sh
vulcan-local -t xxx -checktypes code://./cmd -checktypes code://../other/cmd
```
Perhaps adding a sha of the path or even a sha of the tar file to the image name, or the `_codeN` commented elsewhere.

Reply:
Yes I think this could be a good solution, I maybe like more the one related to the _codeN, but do you think we can do it in a future PR?

Reply:
Yes, we can postpone that.